### ISSUE TITLE
[IDEA] Short cut to wikification

### ISSUE THREAD

**Issue by AnthonyMuscio** [TiddlyWiki/TiddlyWiki5#5110]
**Is your feature request related to a problem? Please describe.**
Macros and wiki text can become quite complex and messy when there is a need to Wikify a variable or macro result before use. The safest use of wikify is at the last moment and closed immediately after. This issue is frequently raised with new and even some advanced issue and commonly causes time wasted and confusion. The need to open an close a wikification also make it difficult to implement in macros and view templates etc.. An inline form as discussed below would permit other possibilities.

**Describe the solution you'd like**
If there were a way to delimit a variable or macro in wikitext or filters that indicated it was to be wikified, returning only text before use, in effect an inline wikification, the block form of the wikify widget can be avoided making wikitext much more readable. In fact perhaps something similar to the substitutions would help eg `$<variableormacro>$` (purely illustrative)

**Describe alternatives you've considered**
- Moving wikify into a macro does not work
- Transclusions and filtered transclusions are unnecessarily complex and need an additional tiddler to work, itself making it more complex.
- Its difficult/impossible to include $wikify in view templates such that the result is available to the wikitext inside the body of the tiddler.

**Additional context**
This issue seems very clear to me with the need to use Wikify just to "condition" a variable so it can be used correctly as unnecessarily complex. I do not have any idea about the appropriate delimiters to use in Wikitext and filters but I am confident this would help users make use of computed variables.

In the TiddlyWiki.com documentation there are many examples of Widgets that use "literal values" but in practice I have always wanted to use a macro variable, field or transclusion in the parameters to widgets, and I would contribute to the documentation examples that are not only using literal parameters, however the need to use Wikify in some cases will make even the documentation less readable, unless we have an inline wikification/evaluation feature to do this.

I believe the way to describe what I am after is to have a macro or variable "evaluated" before use, rather than the wikitext being passed through for subsequent rendering. This feature could be divorced from wikification and provide a more generic solution and that is to be able to delineate something in wikitext to be "evaluated" according to wikitext rules, but returning a simple equivalent to literals for use in wikitext and filters. 

This method if adopted may also provide an alternative to using the text widget to condition a result before display.
`<$text text=<<variableormacro>>/> `could be replaced by `$<variableormacro>$ `(purely illustrative)
- Or perhaps this is two methods, one inside filters and one in wikitext. 

Another thought is could we make use of parenthesis to indicate an order of evaluation?
`[all[current]get[fieldname]match(<varormacroname>)]` such that items in the (is wikified as plain text) before the filter is completed.

I am happy to give more examples but hope this need is clear.

```
<$wikify name="samename" text="""<<samename>>""">

in here <<samename>> uses the same variable or macroname but it has being "evaluated"

</$wikify>
```

Tony

**Comment by AnthonyMuscio** [comment]
I just want to add, I have developed a comprehensive resource to support clarity with many of the TiddlyWiki features. The need to wikify is really an outlier and inconsistent with the other methods available.

See https://anthonymuscio.github.io/#Standard%20Nomenclature

**Comment by AnthonyMuscio** [comment]
In the following notes at https://anthonymuscio.github.io/#%E2%9D%B7%20Widget%20and%20HTML%20attributes%20or%20parameters

You see how we have to say;

attributename=value

A common use is to provide a parameter to the style attribute

- eg style="color: {{!!color}};" this will not work as we are expecting the parameter to be wikified before its use, it will not be.

As a result you need to 
- construct the whole of the attribute value before use.

Question:
Could an extension of this issue allow?;

- `style="color: $({{!!color}})$;"` the color to be evaluated before the attribute is?
- Again illustrative only like `$<variableormacro>$`

**Comment by Jermolene** [comment]
Thanks @AnthonyMuscio.

Just to check, I think the idea here is to introduce a way for attribute values to be wikified before they are used. For example, if we used backticks as the syntax then this:

```
<$text text=`<<version>>`>
```

would give the variable "text" the value "v5.1.23" (etc).

I'd like to understand the use cases better, though, given that there a few existing alternative ways to achieve the same thing.

**Comment by AnthonyMuscio** [comment]
Jeremy

The main use case is to remove the need to use the wikify widget just to clean values for use in filters and other places. The temr I like is to evaluate rather than wikify.

In your example above is it not equivalent to `<$text text=<<version>> >` I am not sure I understand your example.

Here is an example
```
\define new-field(name:"compound") {{!!fieldname}}$name$

new-field=<<new-field>>

<$list filter="[<new-field>]">

</$list>

<$wikify name=new-field text="<<new-field>>">
<$list filter="[<new-field>]">

</$list>
</$wikify>
```

- In the first list above the macro is not evaluated before use in the filter.
- In the second we have to use wikify so the evaluated result can be used in the filter

The following example assumes we had ( ) delimiting a macro or variable name that would first be evaluated removing the need to use wikify.
 
```
<$list filter="[(new-field)]">

</$list>
```

Something similar outside filters would also be desirable, eg using your example
<$set name=new-field value=`<<new-field param>>`>
would evaluate the macro convert it to text and set new-field to a text result.
`<<new-field param>>`

Tony

**Comment by AnthonyMuscio** [comment]
Another thought is how we make use of the filtered transclusions, triple curly braces.

We are often forced to use <$text text={{{ filter }}}/> to display the result in a form that is not subsequently wikified into title links or ignore camel case. So an alternate approach may be wiki-markup that allows the content to be wikified and returned as text only.

A dummy example
Replace <$text text={{{ filter }}}/>
With ( {{{ filter }}} )  to return evaluated text
Replace <<new-field "param">>
with ( <<new-field "param">> ) to return evaluated text

Tony

**Comment by AnthonyMuscio** [comment]
Sorry more has come to mind as a result of items in the forums

The substitutions such as $paramname$ and $(varname)$ do something similar within macros.

Perhaps a form of this could be used for evaluated in filters (not within macros) and in filters and wiki text? 
In many ways this will be "intuitive", make use of existing mark-up standards.

Eg; extend the $(varname)$ to evaluate and replace with text in and out of macros? This will also imply the need for delimiters;

```
param="$(varname)$"

[[$(varname)$]]
```

**Comment by AnthonyMuscio** [comment]
I am yet to test if the new macrocall $output parameter can help here. But some feedback would be appreciated.

Tony

**Comment by Jermolene** [comment]
Hi @AnthonyMuscio

> The main use case is to remove the need to use the wikify widget just to clean values for use in filters and other places. The temr I like is to evaluate rather than wikify.

It is confusing to others if you use different terminology for established concepts within the project.

> In your example above is it not equivalent to `<$text text=<<version>> >` I am not sure I understand your example.

The example I gave could have been rewritten as `<$text text=<<version>> >`, try this one:

```
<$text text=`<$list filter="a b c d"/>`
```

> Here is an example

An example of what?

I'm afraid I found the rest of your comments very hard to follow. I'm not sure if you're asking a question or making a proposal. I don't wish to discourage you from participating but I would appreciate it enormously if you could express your ideas more clearly.

It might help to pay more attention to how others post. I take a lot of time over posting, and take care to try to punctuate correctly, and use clear English that can be followed by our non-English speaking contributors. This reply has taken me approximately 10 minutes to think about and write, which is a noticeable chunk of my available time for working on the project today. We owe it to others to treat each others time as more valuable than our own. In this community as in every community, it really pays to observe how others interact, and to take your cues from the behaviour of others if you are unsure yourself.

**Comment by AnthonyMuscio** [comment]
Jeremy

I am sorry you feel that way and really do not understand how the original thread is not carfuly written and fully illustrated.

I will review at length and try to improve.

In short

Often when we use a macro or variable in a filter we must first wikify it to obtain the result and present the result to the filter. I wonder if we could have a way to provide a variable in a filter such that it is evaluated and returns a string so comparisons will work.

Regards
Tony
